File: android/app/src/main/java/net/ipv64/kivop/dtos/PollServiceDTOs/GetPollResultsDTO.kt
package net.ipv64.kivop.dtos.PollServiceDTOs

import java.util.UUID
import java.time.LocalDateTime

data class GetPollResultsDTO (
    var myVotes : List<UByte>, // empty: did not vote at all
    var totalCount : UInt,
    var identityCount : UInt,
    var results : List<GetPollResultDTO>,
)
data class GetPollResultDTO (
    var index : UByte,
    var text : String,
    var count : UInt,
    var percentage : Double,
    var identities : List<GetIdentityDTO>?,
)

File: DTOs/Sources/PollServiceDTOs/GetPollResultsDTO.swift
import Foundation

public struct GetPollResultsDTO: Codable {
    public var myVotes: [UInt8] // empty: did not vote at all
    public var totalCount: UInt
    public var identityCount: UInt
    public var results: [GetPollResultDTO]
    
    public init(myVotes: [UInt8] = [], totalCount: UInt = 0, identityCount: UInt = 0, results: [GetPollResultDTO] = []) {
        self.myVotes = myVotes
        self.totalCount = totalCount
        self.identityCount = identityCount
        self.results = results
    }
}
public struct GetPollResultDTO: Codable {
    public var index: UInt8
    public var text: String
    public var count: UInt
    public var percentage: Double
    public var identities: [GetIdentityDTO]?
    
    public init(index: UInt8, text: String, count: UInt, percentage: Double, identities: [GetIdentityDTO]? = nil) {
        self.index = index
        self.text = text
        self.count = count
        self.percentage = percentage
        self.identities = identities
    }
}
